### ISSUE TITLE
CMD scripts not recognized, despite present in PATH

### ISSUE THREAD

**Issue by kenorb** [git-for-windows/git#1774]
- [X] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options
git version 2.18.0.windows.1
cpu: x86_64
built from commit: cd1a74fc9dc8a07626c216940db9a51f25206e03
sizeof-long: 4
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.15063]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
$ cat /etc/install-options.txt
Editor Option: VIM
Path Option: CmdTools
SSH Option: OpenSSH
CURL Option: OpenSSL
CRLF Option: LFOnly
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Enabled
```

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

```
$ echo $SHELL
/usr/bin/bash
```

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
$ aws --version
bash: aws: command not found
```

 - What did you expect to occur after running these commands?

```
$ aws.cmd --version
aws-cli/1.15.65 Python/3.6.0 Windows/10 botocore/1.10.64
```

 - What actually happened instead?

The `aws` command cannot be found.

----

Basically `aws` command doesn't exist, but `aws.cmd` does. The command works in other terminals such as Hyper, CMD or PowerShell.

I believe the problem is that this shell doesn't look for scripts with `.cmd` extension.

See:

```
$ whereis aws
C:\Program Files\Amazon\AWSCLI\bin\aws.cmd
$ aws
bash: aws: command not found
$ echo $PATH | egrep ":[^:]+AWS[^:]+:" -o
:/c/Program Files/Amazon/AWSCLI/bin:
$ ls /c/Program\ Files/Amazon/AWSCLI/bin
aws.cmd
```

**Comment by dscho** [comment]
Is the same issue present in [MSYS2](https://msys2.github.io/)? If so, this issue might be better reported to https://github.com/Alexpux/MSYS2-packages, as it would be an MSYS2 Bash issue.

**Comment by marekadare** [comment]
I ran into the same Problem.
The .cmd files are not recognized by bash because they are windows "commands".

I then installed python and pip to install awscli and it worked for me (https://docs.aws.amazon.com/cli/latest/userguide/awscli-install-windows.html#awscli-install-windows-pip)

However I can't get aws-cli to prompt for a MFA Token when I connect to AWS because git-bash has issues to execute python.

I hope this helps :)
